include/ginkgo/core/config/type_descriptor.hpp

```diff
@@ -59,9 +63,15 @@ class type_descriptor final {
      */
     const std::string& get_index_typestr() const;
 
+    /**
+     * Get the global index type string
+     */
+    const std::string& get_global_index_typestr() const;
```

Comment:
Just `index` and `global_index` is a bit confusing IMO. It would be nice to change it to `local_index` and `global_index`, but I understand that that would be a breaking change.